source/components/gitops/Activate.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState } from 'react';
+import { Box, Text } from 'ink';
+import SelectInput from 'ink-select-input';
+import { activateRepo, configurePermit } from '../../lib/gitops/utils.js';
+type GitConfig = {
+	url: string;
+	main_branch_name: string;
+	credentials: {
+		auth_type: string;
+		username: string;
+		private_key: string;
+	};
+	key: string;
+};
+type Props = {
+	accessToken: string;
+	projectKey: string;
+	config: GitConfig;
+	onActivate: (isConfigured: boolean) => void;
+	onError: (error: string) => void;
+};
+type ActivateSelect = {
+	label: string;
+	value: boolean;
+};
+
+const Activate: React.FC<Props> = ({
+	accessToken,
+	projectKey,
+	config,
+	onActivate,
+	onError,
+}) => {
+	const [GitConfigResponse, setGitConfigRespone] = useState<{
+		id: string;
+		status: string;
+		key: string;
+	}>({ id: '', status: '', key: '' });
+	const activateItemSelect = [
+		{
+			label: 'Yes',
+			value: true,
+		},
+		{
+			label: 'No',
+			value: false,
+		},
+	];
+	useEffect(() => {
```

Comment:
This function should be called either from the root component or the previous component. It is against the SOLID principles that a component that calling `activate` is also doing the configuration itself.